agent/consul/internal_endpoint_test.go

```diff
@@ -907,7 +1071,6 @@ service_prefix "db" {
 		req := structs.ServiceSpecificRequest{
 			Datacenter:   "dc1",
 			ServiceName:  "gateway",
-			ServiceKind:  structs.ServiceKindTerminatingGateway,
```

Comment:
If we infer TG vs IG now, should the test somehow verify the _results_ are for TG vs IG since we aren't explicitly requesting it one way or the other?

Reply:
Further down where we check the results, we verify that each GatewayService has `structs.ServiceKindTerminatingGateway` for its GatewayKind, so I think we're already doing this.